src/config.c

```diff
@@ -2150,90 +2155,79 @@ static int updateAppendonly(int val, int prev, const char **err) {
     return 1;
 }
 
-static int updateSighandlerEnabled(int val, int prev, const char **err) {
+static int applySighandlerEnabled(const char **err) {
     UNUSED(err);
-    UNUSED(prev);
-    if (val)
+    if (server.crashlog_enabled)
         setupSignalHandlers();
     else
         removeSignalHandlers();
     return 1;
 }
 
-static int updateMaxclients(long long val, long long prev, const char **err) {
-    /* Try to check if the OS is capable of supporting so many FDs. */
-    if (val > prev) {
```

Comment:
do we know why the old code used to refuse a request to shrink the config? (save memory)

Reply:
see comment below.